sdk/go/kubernetes/helm/v2/chart.go

```diff
@@ -0,0 +1,289 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// *** WARNING: this file was generated by pulumigen. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+package helm
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi-kubernetes/sdk/v2/go/kubernetes/yaml"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+)
+
+// Chart is a component representing a collection of resources described by an arbitrary Helm
+// Chart. The Chart can be fetched from any source that is accessible to the `helm` command
+// line. Values in the `values.yml` file can be overridden using `ChartOpts.values` (equivalent
+// to `--set` or having multiple `values.yml` files). Objects can be transformed arbitrarily by
+// supplying callbacks to `ChartOpts.transformations`.
+//
+// `Chart` does not use Tiller. The Chart specified is copied and expanded locally; the semantics
+// are equivalent to running `helm template` and then using Pulumi to manage the resulting YAML
+// manifests. Any values that would be retrieved in-cluster are assigned fake values, and
+// none of Tiller's server-side validity testing is executed.
+type Chart struct {
+	pulumi.ResourceState
+
+	Resources pulumi.Output
+}
+
+// NewChart registers a new resource with the given unique name, arguments, and options.
+func NewChart(ctx *pulumi.Context,
+	name string, args ChartArgs, opts ...pulumi.ResourceOption) (*Chart, error) {
+
+	// Register the resulting resource state.
+	chart := &Chart{}
+	err := ctx.RegisterComponentResource("kubernetes:helm.sh/v2:Chart", name, chart, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	// Make the component the parent of all subsequent resources.
+	opts = append(opts, pulumi.Parent(chart))
+
+	// Honor the resource name prefix if specified.
+	if args.ResourcePrefix != "" {
+		name = args.ResourcePrefix + "-" + name
+	}
+
+	resources, err := parseChart(ctx, name, args, pulumi.Parent(chart))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse chart")
+	}
+	chart.Resources = resources
+
+	// Finally, register all of the resources found.
+	// Note: Go requires that we "pull" on our futures in order to get them scheduled for execution. Here, we use
+	// the engine's RegisterResourceOutputs to wait for the resolution of all resources that this Helm chart created.
+	err = ctx.RegisterResourceOutputs(chart, pulumi.Map{"resources": resources})
+	if err != nil {
+		return nil, errors.Wrap(err, "registering child resources")
+	}
+
+	return chart, nil
+}
+
+func parseChart(ctx *pulumi.Context, name string, args ChartArgs, opts ...pulumi.ResourceOption,
+) (pulumi.Output, error) {
+
+	resources := args.ToChartArgsOutput().ApplyT(func(args chartArgs) (map[string]pulumi.Resource, error) {
```

Comment:
Total nit - but is it worth having this line in the caller - so that this function can be simpler (just pass a `chartArgs` into this function)?